crates/router/src/core/admin.rs

```diff
@@ -258,6 +260,39 @@ pub async fn create_merchant_account(
         .await
         .to_duplicate_response(errors::ApiErrorResponse::DuplicateMerchantAccount)?;
 
+    // Call to DE here
+    // Check if creation should be based on default profile
+    #[cfg(all(feature = "dynamic_routing", feature = "v1"))]
+    {
+        if state.conf.open_router.enabled {
+            merchant_account
+                .default_profile
+                .as_ref()
+                .async_map(|profile_id| {
+                    let merchant_account_req = open_router::MerchantAccount {
+                        merchant_id: profile_id.get_string_repr().to_string(),
+                        gateway_success_rate_based_decider_input: None,
+                    };
+
+                    let url = format!(
+                        "{}/{}",
+                        &state.conf.open_router.url, "merchant-account/create"
+                    );
+                    routing::helpers::call_decision_engine::<_, String>(
+                        &state,
+                        url,
+                        services::Method::Post,
+                        Some(merchant_account_req),
+                        "decision_engine_merchant_account_create".to_string(),
+                    )
+                })
+                .await
+                .transpose()
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Failed to create merchant account on decision engine")?;
+        }
```

Comment:
```suggestion
       routing::helpers::create_merchant_account_in_decision_engine(&state, merchant_account).await?
```
handle all the conditional logics inside the DE utils